From 4d77b8ba4c33af64e75b9f1b8fef874369bd855b Mon Sep 17 00:00:00 2001
From: Harry Wentland <harry.wentland@amd.com>
Date: Tue, 9 May 2017 10:41:42 -0400
Subject: [PATCH 1637/2094] drm/amd/display: Keep DVI_SINGLE_LINK signal if low
 clk

If user is using DVI->HDMI dongle dual link signal might pose a
problem. Keep single link signal type if clk is lower than
max tmds clk.

Signed-off-by: Harry Wentland <harry.wentland@amd.com>
---
 drivers/gpu/drm/amd/display/dc/core/dc_resource.c | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/amd/display/dc/core/dc_resource.c b/drivers/gpu/drm/amd/display/dc/core/dc_resource.c
index ae8b221..9588217 100644
--- a/drivers/gpu/drm/amd/display/dc/core/dc_resource.c
+++ b/drivers/gpu/drm/amd/display/dc/core/dc_resource.c
@@ -1283,9 +1283,12 @@ static void update_stream_signal(struct core_stream *stream)
 		stream->signal = stream->public.output_signal;
 	}
 
-	if (stream->signal == SIGNAL_TYPE_DVI_SINGLE_LINK &&
-		stream->public.timing.pix_clk_khz > TMDS_MAX_PIXEL_CLOCK_IN_KHZ)
-		stream->signal = SIGNAL_TYPE_DVI_DUAL_LINK;
+	if (dc_is_dvi_signal(stream->signal)) {
+		if (stream->public.timing.pix_clk_khz > TMDS_MAX_PIXEL_CLOCK_IN_KHZ)
+			stream->signal = SIGNAL_TYPE_DVI_DUAL_LINK;
+		else
+			stream->signal = SIGNAL_TYPE_DVI_SINGLE_LINK;
+	}
 }
 
 bool resource_is_stream_unchanged(
-- 
2.7.4

